mentoring/dashboard.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2015 Harvard, edX & OpenCraft
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Dashboard: Summarize the results of self-assessments done by a student with Problem Builder MCQ
+blocks.
+
+The author of this block specifies a list of Problem Builder blocks containing MCQs. This block
+will then display a table summarizing the values that the student chose for each of those MCQs.
+"""
+# Imports ###########################################################
+
+import json
+import logging
+
+from .dashboard_visual import DashboardVisualData
+from .mcq import MCQBlock
+from .sub_api import sub_api
+from webob import Response
+from xblock.core import XBlock
+from xblock.exceptions import XBlockNotFoundError
+from xblock.fields import Scope, Dict, List, String
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.helpers import child_isinstance
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+def _(text):
+    """ A no-op to mark strings that we need to translate """
+    return text
+
+# Classes ###########################################################
+
+
+@XBlock.needs("i18n")
+class DashboardBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    A block to summarize self-assessment results.
+    """
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Display name for this module"),
+        scope=Scope.settings,
+        default=_('Self-Assessment Summary'),
+    )
+    mentoring_ids = List(
```

Comment:
Putting in valid JSON not of the right value-- like an integer, produces an error, but not a user friendly one. It's the generic studio error and not the xblock notification error, I think.